aiida_quantumespresso/workflows/ph/base.py

```diff
@@ -22,152 +21,126 @@ class PhBaseWorkChain(BaseRestartWorkChain):
+            self.ctx.inputs.parent_folder = self.ctx.restart_calc.outputs.remote_folder
 
-@register_error_handler(PhBaseWorkChain, 400)
-def _handle_fatal_error_read_namelists(self, calculation):
-    """
-    The calculation failed because it could not read the generated input file
-    """
-    if any(['reading inputph namelist' in w for w in calculation.res.warnings]):
-        self.report('PhCalculation<{}> failed because of an invalid input file'.format(calculation.pk))
-        return ErrorHandlerReport(True, True, self.exit_codes.ERROR_CALCULATION_INVALID_INPUT_FILE)
+    def report_error_handled(self, calculation, action):
+        """Report an action taken for a calculation that has failed.
 
+        This should be called in a registered error handler if its condition is met and an action was taken.
 
-@register_error_handler(PhBaseWorkChain, 300)
-def _handle_error_exceeded_maximum_walltime(self, calculation):
-    """
-    Calculation ended nominally but ran out of allotted wall time
-    """
-    if 'Maximum CPU time exceeded' in calculation.res.warnings:
-        self.ctx.restart_calc = calculation
-        self.report(
-            'PhCalculation<{}> terminated because maximum wall time was exceeded, restarting'.format(calculation.pk)
-        )
-        return ErrorHandlerReport(True, True)
+        :param calculation: the failed calculation node
+        :param action: a string message with the action taken
+        """
+        arguments = [calculation.process_label, calculation.pk, calculation.exit_status, calculation.exit_message]
+        self.report('{}<{}> failed with exit status {}: {}'.format(*arguments))
+        self.report('Action taken: {}'.format(action))
 
 
-@register_error_handler(PhBaseWorkChain, 200)
-def _handle_fatal_error_not_converged(self, calculation):
-    """
-    The calculation failed because it could not read the generated input file
-    """
-    if 'Phonon did not reach end of self consistency' in calculation.res.warnings:
-        alpha_mix_old = calculation.inputs.parameters.get_dict()['INPUTPH'].get('alpha_mix(1)', self.defaults.alpha_mix)
-        alpha_mix_new = 0.9 * alpha_mix_old
-        self.ctx.inputs.parameters['INPUTPH']['alpha_mix(1)'] = alpha_mix_new
+@register_error_handler(PhBaseWorkChain, 600)
+def _handle_unrecoverable_failure(self, calculation):
+    """Calculations with an exit status below 400 are unrecoverable, so abort the work chain."""
+    if calculation.exit_status < 400:
+        self.report_error_handled(calculation, 'unrecoverable error, aborting...')
+        return ErrorHandlerReport(True, True, self.exit_codes.ERROR_UNRECOVERABLE_FAILURE)
+
+
+@register_error_handler(PhBaseWorkChain, 580)
+def _handle_out_of_walltime(self, calculation):
+    """In the case of `ERROR_OUT_OF_WALLTIME` calculation shut down neatly and we can simply restart."""
+    if calculation.exit_status == PhCalculation.spec().exit_codes.ERROR_OUT_OF_WALLTIME.status:
```

Comment:
I believe you can just say `PhCalculation.exit_codes` if you want (without `spec()`).

Reply:
correct, will adapt